app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -19,14 +19,28 @@ def pvm_instance(instance_id)
     pvm_instances_api.pcloud_pvminstances_get(cloud_instance_id, instance_id)
   end
 
-  def image(img_id)
-    images_api.pcloud_cloudinstances_images_get(cloud_instance_id, img_id)
+  def image(image_id)
+    image = images.detect { |img| img.image_id == image_id }
```

Comment:
I'm not sure if the `detect` is the best lookup method here, but it will return `nil` if the image already already been fetched and saved to `images` by the `GET /images` call.

Reply:
Better than looping through the array each time would be to store the images as a hash with the image_id as the hash key:
```
def images_by_id
  @images_by_id ||= images_api.pcloud_cloudinstances_images_getall(cloud_instance_id).images.index_by(&:image_id)
end

def images
  images_by_id.values
end
```